src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -3028,6 +3028,37 @@ operand immLOffset()
   interface(CONST_INTER);
 %}
 
```

Comment:
I think it will be better to introduce one new immIScale which combines all three.
Then we can combine the newly-added instructs and reduce redundancy. If you do that, you will also need a new enc_class for distinguishing between the different Zba shift instruction encodings.

// Scale/Shift values for Zba instructions
operand immIScale()
%{
predicate(0 <= n->get_int() && (n->get_int() <= 3));
match(ConI);
op_cost(0);
format %{ %}
interface(CONST_INTER);
%}

Reply:
Thank you for the suggestion. It's maybe better to call the `shadd` under macroAssembler (where t0 is actually not used) to avoid the reduction definition of enc_class. New commits pushed.